app/src/main/java/ch/epfl/culturequest/SettingsActivity.java

```diff
@@ -177,7 +190,35 @@ public void selectProfilePicture(View view) {
     }
 
     private void openGallery() {
-        profilePictureSelector.launch(new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI));
+        // start the gallery activity to select a picture with result code TAKE_PICTURE
+        startActivityForResult(new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI), TAKE_PICTURE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
This method looks very similar to the one in ProfileCreationActivity, you should consider refactoring